### PR TITLE
fix issues

### PR DESCRIPTION
**Pull request by MimiYuchenGuo**
#30

### CODE REVIEW

File: tests/testthat/test-getLargeScaleCharacteristics.R [hunk 1]
```diff
@@ -42,17 +44,6 @@ test_that("check getLargeScaleCharacteristics inputs checks", {
     tablesToCharacterize = c("drug_exposure")
   ))
```
Comment (catalamarti) [inline tests/testthat/test-getLargeScaleCharacteristics.R:46]: why these tests were deleted?

File: R/supressCount.R [hunk 2]
```diff
@@ -52,52 +46,36 @@ supressCount <- function(result,
...
+    dplyr::mutate(obscured_ind = dplyr::if_else(
+      .data$estimatesToObscure < .env$minimumCellCount, TRUE, FALSE
+    )) %>%
+    dplyr::mutate(estimatesToObscure = dplyr::if_else(.data$obscured_ind,
+      paste0("<", minimumCellCount),
+      as.character(.data$estimatesToObscure)
+    )) %>%
+    dplyr::rename(!!estimatesToObscure := estimatesToObscure) %>% dplyr::select(-c("obscured_ind"))
+
+  result <- result %>% dplyr::select(-tidyselect::all_of(estimatesToObscure)) %>%
```
Comment (catalamarti) [inline R/supressCount.R]: you can use dplyr::all_of()
and then we have one dependency less

File: R/supressCount.R [hunk 3]
```diff
@@ -17,33 +17,27 @@
...
 supressCount <- function(result,
-                         minimumCellCounts = 5,
-                         globalVariables = c(
-                           "number_observations", "number_subjects"
-                         ),
-                         estimatesToObscure = "count") {
-
+                         minimumCellCount = 5,
+                         cohort_definition_id = NULL,
+                         estimatesToObscure = "concept_count") {
```
Comment (catalamarti) [inline R/supressCount.R]: Is denominator_count also obscured?

File: R/supressCount.R [hunk 4]
```diff
@@ -52,52 +46,36 @@ supressCount <- function(result,
...
-  result$value[
-    result$cohort_definition_id %in% obscured_cohort &
-      result$variable %in% globalVariables
-  ] <- paste0("<", minimumCellCounts)
+  if (!is.null(cohort_definition_id)) {
+    result_to_obscure <- result %>%
+      dplyr::filter(cohort_definition_id %in% .env$cohort_definition_id)
+  }else{result_to_obscure <- result}
+
+  result_to_obscure <- result_to_obscure %>% dplyr::rename(estimatesToObscure := !!estimatesToObscure)
```
Comment (catalamarti) [inline R/supressCount.R]: why is this necessary?

**Review by catalamarti — CHANGES_REQUESTED** [review]
@MimiYuchenGuo some minor changes

@KimLopezGuell  could you please add a test to make sure that the error that you found does not appear again?

File: tests/testthat/test-getLargeScaleCharacteristics.R (hunk 1 above)
Comment (MimiYuchenGuo) [reply tests/testthat/test-getLargeScaleCharacteristics.R:46]: > why these tests were deleted?

cuz I edited the mock function to contain all cdm tables now, previously measurement and device_exposure tables are not in cdm without user specify

File: R/supressCount.R (hunk 3 above)
Comment (MimiYuchenGuo) [reply R/supressCount.R]: > Is denominator_count also obscured?

User can set estimatesToObscure themselves, and yes denominator_count was obscured in https://github.com/oxford-pharmacoepi/LargeScaleCharacteristics/blob/271f499370bcabf1cc3471ce867f6d2f0fdd3940/R/getLargeScaleCharacteristics.R#L447

File: R/supressCount.R (hunk 4 above)
Comment (MimiYuchenGuo) [reply R/supressCount.R]: > why is this necessary?

if they only want to suppress one cohort definition id?

File: tests/testthat/test-getLargeScaleCharacteristics.R (hunk 1 above)
Comment (catalamarti) [reply tests/testthat/test-getLargeScaleCharacteristics.R:46]: Can you then add 
cdm$drug_exposure <- NULL
and see that the error appears?

**Comment by MimiYuchenGuo** [comment]
> @MimiYuchenGuo some minor changes
> 
> @KimLopezGuell could you please add a test to make sure that the error that you found does not appear again?

@KimLopezGuell  There is a test here for target cohort id but please kindly add here more tests if necessary, many thanks
https://github.com/oxford-pharmacoepi/LargeScaleCharacteristics/blob/54673cb0fe224bbf35014d072cd1e087d47cbe9b/tests/testthat/test-getLargeScaleCharacteristics.R#L414
